app/models/businessactivities/TransactionType.scala

```diff
@@ -65,6 +66,30 @@ object TransactionTypes {
     }
   }
 
+  val oldTransactionTypeReader: Reads[Option[TransactionTypes]] =
+    (__ \ "isRecorded").read[Boolean] flatMap {
+      case true => (__ \ "transactions").read[Set[String]].flatMap {x:Set[String] =>
+        x.map {
+          case "01" => constant(Paper) map identity[TransactionType]
+          case "02" => constant(DigitalSpreadsheet) map identity[TransactionType]
+          case "03" =>
+            (__ \ "digitalSoftwareName").read[String].map (DigitalSoftware.apply  _) map identity[TransactionType]
+          case _ =>
+            Reads(_ => JsError((__ \ "transactions") -> play.api.data.validation.ValidationError("error.invalid")))
+        }.foldLeft[Reads[Set[TransactionType]]](
+          Reads[Set[TransactionType]](_ => JsSuccess(Set.empty))
+        ){
+          (result, data) =>
+            data flatMap {m =>
+              result.map {n =>
```

Comment:
Did you decide against re-writing this in a more readable way? :)

Reply:
I think when we were chatting about I was referring to the JSON readers for the [equivalent type in the backend microservice](https://github.com/hmrc/amls/blob/master/app/models/fe/businessactivities/TransactionRecord.scala) but this one can definitely be rewritten in a similar fashion. Would you mind if I put it in a separate pull request so that this can be tested in the meantime?

Reply:
Yep.